xdsl/irdl/declarative_assembly_format.py

```diff
@@ -119,6 +134,32 @@ def parse(
             result_types=result_types, operands=operands, attributes=state.attributes
         )
 
+    def resolve_operand_types(self, state: ParsingState) -> None:
+        for i, operand_type in enumerate(state.operand_types):
+            if operand_type is None:
+                state.operand_types[i] = self._resolve_type(
```

Comment:
is this legal? the dict is being modified while iterated, I don't think the keys are copied by default

Reply:
state.operand_types is a list, and I'm only updating existing elements

Reply:
Ah, my bad, I think this is still not allowed, I'd recommend iterating over the indices and getting the element in the loop

Reply:
Looks like it is allowed all good

Reply:
🫡🚔